### ISSUE TITLE
Media Library no longer allows the ability to view images attached to post

### ISSUE THREAD

**Issue by jamesmrobinson** [WordPress/gutenberg#12753]
When viewing the media library from the classic editor there is the ability to filter results by media type as well as view images attached to a post, unattached, or belonging to the user.

With Gutenberg this is lost and it seems you can only filter by date.

Installing the classic editor plugin restores the Media Library modal view along with the additional filter.

The desired solution would be to restore these filters in Gutenberg.

**Comment by cagivacode** [comment]
I just found this as well......I have thousands of images and trying to find the right one for featured image (old posts did not use them) is a nightmare without this filter

**Comment by tiagogon** [comment]
+1

**Comment by kylemcph** [comment]
Also a report of this from 11543319-hc

**Comment by finalwebsites** [comment]
Same here for a custom post type. With Gutenberg installed I can see the linked posts only via the media library.
It would be great to have a "Media" button, without the need to add a media block first.

**Comment by Wadowad** [comment]
+1

**Comment by yoanmarchal** [comment]
since à long time i use media attachments in order to do "custom slider" but with gutenberg i have no idea how to do that anymore

**Comment by RoqueCampos** [comment]
I'm having the same issue because of that +1

**Comment by venture89** [comment]
Yes, this is a pain to go through so many images to find the ones that are uploaded to the images. Usually, writers upload images and editor's need to find them to set them as featured images...etc. Please fix this bug.

**Comment by venture89** [comment]
> Installing the classic editor plugin restores the Media Library modal view along with the additional filter.

This is no longer a valid solution. Even having the classic editor plugin does not allow you to filter media by 'attached to' post.

**Comment by Drivingralle** [comment]
+1 to get this back

**Comment by tiagogon** [comment]
+1+1+1

**Comment by spencerberry** [comment]
Worth noting here that some work would need to be done to ensure Gutenberg added images are properly seen as attached.

Right now, you can have an image in a Gutenberg post or page but the status in WP Admin > Media will still show unattached.

**Comment by Bergzeit** [comment]
+1

**Comment by simonPaulBergzeit** [comment]
+1

**Comment by Ezokia** [comment]
+1 !!

**Comment by johnbillion** [comment]
Note: This feature is present for the Featured Image control, but not when inserting media using one of the media block types.

**Comment by fylgjur** [comment]
User requesting this feature to return in 3021542-hc

**Comment by tiagogon** [comment]
Great news, @fylgjur :) these feature will be critical for some of the WP installations I have. 

Do you know in which version of WordPress will the fix be?

**Comment by torounit** [comment]
In case of Gallery Block or FeaturedImage , Filter is displayed. 

Occurs only in Image Block.

**Comment by dolgelukkig** [comment]
User asking why this feature is not working anymore in chat #27053515-hc 

The tab is there in de pop up but it's not clickable. 
I tested in a simple site and atomic site, with different blocks (gallery, tiled gallery, image) and the issue is the same in all setups. 

screenshot: https://d.pr/i/8i6yCL

**Comment by paaljoachim** [comment]
Hi @nosquirrelbones James and others could we get a status update on this issue?
Thanks!

**Comment by Soean** [comment]
I can reproduce this issue with Gutenberg 9.9.1

**Comment by tiagogon** [comment]
+1 

_Two years after the launch of a new editor's interface, Wordpress still didn't ensure feature paring with the old version: forcing 5 million installations still using the old editor with a plugin. Why?_

**Comment by Soean** [comment]
> _Two years after the launch of a new editor's interface, Wordpress still didn't ensure feature paring with the old version: forcing 5 million installations still using the old editor with a plugin. Why?_

@tiagogon Fortunately WordPress is an open source project. You are welcome to contribute to it. So that we can reach this goal soon. Thanks!

**Comment by paaljoachim** [comment]
Could we get an update with screenshots comparing the two versions? 
That is a quicker way to get an overview. Thank you!

**Comment by Soean** [comment]
Insert block -> click on "Media library" in the placeholder.

Image, just the date filter
<img width="349" alt="Bildschirmfoto 2021-02-10 um 14 35 42" src="https://user-images.githubusercontent.com/695201/107517274-d90db300-6bad-11eb-9142-2682ee498b70.png">

Gallery, additional filter for attached images
<img width="548" alt="Bildschirmfoto 2021-02-10 um 14 36 02" src="https://user-images.githubusercontent.com/695201/107517270-d7dc8600-6bad-11eb-903d-7902c343ee8d.png">

**Comment by sbathompson-he** [comment]
Helped a client recently who offered a work-around:

Directly Quoted
>Steps:
>1. find unattached media and copy the name
>2. click "attach" and paste in the media name in the search
>3. it will show you the writings it is embedded in regardless of whether it is a post or page

if not found it really is unattached

**Comment by dnnsjsk** [comment]
For anyone looking for a solution for this, check my other comment out:
https://github.com/WordPress/gutenberg/issues/19851#issuecomment-866852934

**Comment by andrewserong** [comment]
I've opened a tentative PR that exposes the additional media library filters when using the media library with the image block. Ready for review / feedback over in https://github.com/WordPress/gutenberg/pull/65965

That PR only exposes the filters for now, it doesn't address one of the issues mentioned earlier in the discussion here (https://github.com/WordPress/gutenberg/issues/12753#issuecomment-556149071):

> Worth noting here that some work would need to be done to ensure Gutenberg added images are properly seen as attached.
> Right now, you can have an image in a Gutenberg post or page but the status in WP Admin > Media will still show unattached.

So, separately to that PR, it'd be good to investigate the desired behaviour for attaching images to a post in Gutenberg. I.e. just one idea — when we add an image to a post, if it isn't already attached to the post then update the image to be attached to the current post id. If it's already attached, then don't switch the attachment. Curious to hear if anyone had other ideas about how that part of things should work!

**Comment by andrewserong** [comment]
Update: I've merged in #65965 which exposes the filters within the media library in the block editor.

For the problem of images that already exist in the media library and are then added to a post or page, but do not become "attached" to that post, I've opened a separate issue over in https://github.com/WordPress/gutenberg/issues/66663 where the discussion can continue.

Currently in Gutenberg, if you upload directly to the post (i.e. drag an image onto the Image block), then it becomes attached to the post. However this does not happen if a user selects the image from the media library.
